blockchain/blockchain.go

```diff
@@ -374,7 +378,11 @@ func (bc *BlockChain) prefetchTxWorker(index int) {
 
 	logger.Debug("prefetchTxWorker is started", "index", index)
 	for followup := range bc.prefetchTxCh {
-		stateDB, err := state.NewForPrefetching(bc.CurrentBlock().Root(), bc.stateCache, bc.snaps)
+		var snaps *snapshot.Tree
+		if bc.cacheConfig.TrieNodeCacheConfig.UseSnapshotForPrefetch {
+			snaps = bc.snaps
+		}
```

Comment:
How about setting `snaps` before the `for` loop?

Reply:
Updated, thanks.